src/App.tsx

```diff
@@ -1,24 +1,150 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { TodoForm } from './components/TodoForm';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { ErrorMessage } from './components/ErrorMessage';
+import { Todo } from './types/Todo';
+import { FilterOptions } from './types/FilterOptions';
+import * as postService from './api/todos';
+import { ErrorMessages } from './types/errorMessages';
+import { prepareTodos } from './utils/prepareTodos';
 
-const USER_ID = 0;
+const USER_ID = 11272;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterOption, setFilterOption] = useState(FilterOptions.All);
+  const [deletingTodoIds, setDeletingTodoIds] = useState<number[]>([]);
+  const [isDisabled, setIsDisabled] = useState(false);
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [hasError, setHasError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(ErrorMessages.None);
+
+  const preparedTodos = useMemo(() => {
+    return prepareTodos(todos, filterOption);
```

Comment:
Consider renaming this function to show that it's an action
```suggestion
    return getPreparedTodos(todos, filterOption);
```